corehq/apps/case_importer/do_import.py

```diff
@@ -250,6 +259,17 @@ def _populate_updated_fields(self, row):
             elif update_value is not None:
                 update_value = _convert_field_value(update_value)
 
+            if update_field_name in self.validated_fields:
+                case_property = self.validated_fields[update_field_name]
+                if not case_property.valid_value(update_value):
+                    # note: probably want to accumulate full list of invalid
+                    # columns for this row and raise an exception with that
+                    # full set of info rather than bailing out on first one
+                    # as we're currently doing here.
+                    if case_property.data_type == 'date':
+                        raise exceptions.InvalidDate(update_field_name)
```

Comment:
What message ultimately gets displayed to the user?

Reply:
![image](https://user-images.githubusercontent.com/430048/124663312-ea669d00-de77-11eb-8cf5-ccbbdaa22990.png)

This is an example of the message that is eventually given to the user. This `InvalidDate` exception already existed in the code but was apparently unused.

Reply:
nice, that looks good